code/modules/reagents/machinery/alchemy.dm

```diff
@@ -77,39 +91,222 @@
+	. = ..()
+	update_icon()
+
+/obj/item/storage/pouch/alchemy/New()
+	. = ..()
+	update_icon()
 
-/obj/item/alchemy/recipe_scroll
-	name = "alchemy recipe"
-	desc = "A tatty paper written in a strange language."
+/obj/item/storage/pouch/alchemy/update_icon()
+	..()
+	cut_overlays()
+	if(contents.len)
+		add_overlay(image('icons/obj/chemical.dmi', "flask_[contents.len]"))
+	if(open)
+		add_overlay(image('icons/obj/chemical.dmi', "open_lid"))
+	else
+		add_overlay(image('icons/obj/chemical.dmi', "closed_lid"))
+
+/obj/item/storage/pouch/alchemy/can_interact(mob/user)
+	if((!ishuman(user) && (loc != user)) || user.stat || user.restrained())
+		return 1
+	if(istype(loc, /obj/item/storage))
+		return 2
+	return 0
+
+/obj/item/storage/pouch/alchemy/open(mob/user)
+	if(!open)
+		to_chat(user, SPAN_NOTICE("\The [src] is closed."))
+		return
+	. = ..()
+
+/obj/item/storage/pouch/alchemy/verb/quick_open_close(mob/user)
+	set name = "Close Satchel"
+	set category = "Object"
+	set src in view(1)
+
+	if(!isliving(loc))
+		return
+
+	if(can_interact(user) == 1)
+		return
+
+	open_close(user)
+
+/obj/item/storage/pouch/alchemy/AltClick(mob/user)
+	var/able = can_interact(user)
+	if(able == 1)
+		return
+
+	if(able == 2)
+		to_chat(user, SPAN_NOTICE("You cannot open \the [src] while it\'s in a container."))
+		return
+
+	open_close(user)
+
+/obj/item/storage/pouch/alchemy/proc/open_close(mob/living/carbon/human/H, user)
+	if(!open)
+		to_chat(user, SPAN_NOTICE("You open \the [src]."))
+		w_class = ITEM_SIZE_BULKY
+		open = TRUE
+	else
+		to_chat(user, SPAN_NOTICE("You close \the [src]."))
+		w_class = ITEM_SIZE_NORMAL
+		open = FALSE
+	playsound(loc, 'sound/items/storage/briefcase.ogg', 100, 1)
+	update_icon()
+
+/obj/item/storage/pouch/alchemy/attackby(obj/item/W, mob/user)
+	if(!open)
+		to_chat(user, SPAN_NOTICE("You try to access \the [src] but it\'s closed!"))
+		return
+	. = ..()
+
+/obj/item/storage/pouch/alchemy/attack_self(mob/user)
+	if(can_interact(user) == 1)
+		return
+	open_close(user)
+
+// FIXME: With sliding behavior turned on by default, you can still draw items from it while closed, please fix
+/obj/item/storage/pouch/alchemy/attack_hand(mob/living/carbon/human/user)
+	if(sliding_behavior && contents.len && (src in user))
+		var/obj/item/I = contents[contents.len]
+		if(istype(I) && src.open)
+			hide_from(usr)
+			var/turf/T = get_turf(user)
+			remove_from_storage(I, T)
+			usr.put_in_hands(I)
+			add_fingerprint(user)
+	..()
+
+// A paper containing all alchemy recipes so that alchemists have at least a basis of knowledge on chemical reactions
+// And they don't have to codedive to get to know them.
+/obj/item/paper/alchemy_recipes
+	name = "Alchemy Recipes"
+	desc = "A tattered parchment written in a strange language, detailing the sum knowledge of alchemical compounds..."
 	icon = 'icons/obj/scroll_bandange.dmi'
 	icon_state = "Scroll ink"
+	item_state = "crayon_scroll_open"
+	language = LANGUAGE_CULT // So that only cultists can read it
```

Comment:
Poor hunters

Reply:
Read the PR, this removes Hunter's free Alchemy perk